packages/uploadthing/src/internal/types.ts

```
@@ -156,8 +163,12 @@ export type UploadBuilderDef<TParams extends AnyParams> = {
   routerConfig: FileRouterInputConfig;
   routeOptions: RouteOptions;
   inputParser: JsonParser;
-  // eslint-disable-next-line @typescript-eslint/ban-types
-  middleware: MiddlewareFn<TParams["_input"], {}, TParams["_middlewareArgs"]>;
+  middleware: MiddlewareFn<
+    TParams["_input"]["out"],
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    {},
```

Biome

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)